components/dashboard/src/repositories/detail/prebuilds/PrebuildDetailPage.tsx

```diff
@@ -0,0 +1,148 @@
+    const { repositoryId, prebuildId } = useParams<PageRouteParams>();
+
+    const { data: repository, isLoading: repositoryIsLoading } = useConfiguration(repositoryId);
+    const { toast } = useToast();
+
+    // TODO: watch
+    const { data: prebuild, isLoading: prebuildIsLoading } = usePrebuildQuery(prebuildId);
+
+    const { emitter: logEmitter } = usePrebuildLogsEmitter(prebuildId);
+
+    useEffect(() => {
+        logEmitter.on("error", (err: Error) => {
+            toast("Failed to fetch logs: " + err.message);
+        });
+    }, [logEmitter, toast]);
+
+    const prebuildPhase = useMemo(() => {
+        switch (prebuild?.status?.phase?.name) {
+            case PrebuildPhase_Phase.QUEUED:
+                return {
+                    icon: <Loader2Icon size={20} className="text-gray-500 animate-spin" />,
+                    description: "Prebuild queue",
+                };
+            case PrebuildPhase_Phase.BUILDING:
+                return {
+                    icon: <Loader2Icon size={20} className="text-gray-500 animate-spin" />,
+                    description: "Prebuild building",
+                };
+            case PrebuildPhase_Phase.ABORTED:
+                return {
+                    icon: <CircleSlash2Icon size={20} className="text-gray-500" />,
+                    description: "Prebuild aborted",
+                };
+            case PrebuildPhase_Phase.TIMEOUT:
+                return {
+                    icon: <CircleSlash2Icon size={20} className="text-gray-500" />,
+                    description: "Prebuild timeout",
+                };
+            case PrebuildPhase_Phase.AVAILABLE:
+                return {
+                    icon: <CheckCircle2Icon size={20} className="text-green-500" />,
+                    description: "Prebuild available",
+                };
+            case PrebuildPhase_Phase.FAILED:
+                return {
+                    icon: <AlertTriangleIcon size={20} className="text-kumquat-base" />,
+                    description: "Prebuild available",
+                };
+
+            case PrebuildPhase_Phase.UNSPECIFIED:
+            default:
+                return {
+                    icon: <CircleSlash2Icon size={20} className="text-gray-500" />,
+                    description: "Prebuild unknown",
+                };
+        }
+    }, [prebuild?.status?.phase?.name]);
+
+    return (
+        <div className="w-full">
+            <BreadcrumbNav
+                pageTitle="Prebuild history"
+                pageDescription={
+                    <>
+                        <span className="font-semibold">{repository?.name || ""}</span>{" "}
+                        <span className="text-pk-content-secondary">{prebuild?.ref || ""}</span>
+                    </>
+                }
+                backLink="/repositories"
+            />
+            {(repositoryIsLoading || prebuildIsLoading) && <Loader2 className="animate-spin" />}
```

Comment:
👀 we can use https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/components/podkit/loading/LoadingState.tsx

Reply:
My knowledge about dashboard is out-of-date (